ddtrace/_logger.py

```diff
@@ -0,0 +1,64 @@
+import logging
+from logging.handlers import RotatingFileHandler
+import os
+
+from ddtrace.internal.utils.formats import asbool
+
+
+DEFAULT_FILE_SIZE_BYTES = 15 << 20  # 15 MB
+
+
+def configure_ddtrace_logger():
+    # type: () -> None
+    """Configures ddtrace log levels and file paths.
+
+    Customization is possible with the environment variables:
+        ``DD_TRACE_DEBUG``, ``DD_TRACE_LOG_FILE_LEVEL``, and ``DD_TRACE_LOG_FILE``
+
+    By default, when none of the settings have been changed, ddtrace loggers
+        inherit from the root logger in the logging module and no logs are written to a file.
+
+    When DD_TRACE_DEBUG has been enabled:
+        - Propagate logs up so logs show up in the application logs if a file path wasn't provided
+        - Logs are routed to a file when DD_TRACE_LOG_FILE is specified, using the log level in DD_TRACE_LOG_FILE_LEVEL.
+        - Child loggers inherit from the parent ddtrace logger
+
+    Note(s):
+        1) The ddtrace-run logs under commands/ddtrace-run do not follow DD_TRACE_LOG_FILE if DD_TRACE_DEBUG is enabled.
```

Comment:
Ugh this is a pain but makes sense after reading the code. Can we elaborate just a tad to say that a basicConfig() call is made in `ddtrace_run` when `DD_TRACE_DEBUG` is enabled and so the logging configuration we do here isn't applied.

Reply:
In the latest commit, I wrote:
```
        1) The ddtrace-run logs under commands/ddtrace_run do not follow DD_TRACE_LOG_FILE if DD_TRACE_DEBUG is enabled.
            This is because ddtrace-run calls ``logging.basicConfig()`` when DD_TRACE_DEBUG is enabled, so
            this configuration is not applied.
```

Let me know if that clarifies it a bit!

Reply:
Marking this as resolved since I changed the doc string.